pkgs/unified_analytics/lib/src/asserts.dart

```diff
@@ -122,15 +145,19 @@ void checkBody(Map<String, Object?> body) {
     // GA4 Limitation:
     // User property names must be 24 characters or fewer
     if (key.length > 24) {
-      throw AnalyticsException('Limit user property names to 24 chars or less');
+      throw AnalyticsException('Limit user property names to 24 chars or less\n'
+          'The user property key: "$key" is too long');
     }
 
     // GA4 Limitation:
     // User property values must be 36 characters or fewer
     final userPropValue = value['value'];
     if (userPropValue is String && userPropValue.length > 36) {
       throw AnalyticsException(
-          'Limit user property values to 36 chars or less');
+        'Limit user property values to 36 chars or less\n'
+        'The value for the user property: "$key" which has '
+        'a value: "${value['value']}" is too long',
```

Comment:
slight reword for clarity:

```suggestion
        'For the user property key "$key", the value "$value" is too long',
```